File: pages/api/health.ts
import type { NextApiRequest, NextApiResponse } from 'next';

type HealthResponse = {
  status: string;
  gemini_configured: boolean;
};

export default function handler(
  req: NextApiRequest,
  res: NextApiResponse<HealthResponse>
) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  
  if (req.method !== 'GET') {
    res.status(405).end();
    return;
  }

  res.status(200).json({
    status: 'healthy',
    gemini_configured: !!process.env.GEMINI_API_KEY,
  });
}

File: api/health.py
from http.server import BaseHTTPRequestHandler
import json
import os

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        
        response = {
            'status': 'healthy',
            'gemini_configured': bool(os.getenv('GEMINI_API_KEY'))
        }
        
        self.wfile.write(json.dumps(response).encode('utf-8'))
